src/node_file.cc

```diff
@@ -885,6 +885,54 @@ static void WriteBuffer(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+// Wrapper for writev(2).
+//
+// bytesWritten = writev(fd, chunks, position, callback)
+// 0 fd        integer. file descriptor
+// 1 chunks    array of buffers to write
+// 2 position  if integer, position to write at in the file.
+//             if null, write from the current position
+static void WriteBuffers(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  CHECK(args[0]->IsInt32());
+  CHECK(args[1]->IsArray());
+
+  int fd = args[0]->Int32Value();
+  Local<Array> chunks = args[1].As<Array>();
+  int64_t pos = GET_OFFSET(args[2]);
+  Local<Value> req = args[3];
+
+  uint32_t chunkCount = chunks->Length();
+
+  uv_buf_t* iovs = new uv_buf_t[chunkCount];
+
+  for (uint32_t i = 0; i < chunkCount; i++) {
+    Local<Value> chunk = chunks->Get(i);
+
+    if (!Buffer::HasInstance(chunk)) {
+      delete[] iovs;
+      return env->ThrowTypeError("Array elements all need to be buffers");
+    }
+
+    const char* buf = Buffer::Data(chunk);
+    size_t len = Buffer::Length(chunk);
+
+    iovs[i] = uv_buf_init(const_cast<char *>(buf), len);
+  }
+
+  if (req->IsObject()) {
+    ASYNC_CALL(write, req, fd, iovs, chunkCount, pos)
```

Comment:
It's used to check if it's an object, then passed. Sure, I could remove it, but how would that help? I doubt the compiler isn't smart enough to optimize this one way or another if you think this is inefficient.